tests/integration/pilot/mcs/discoverability/discoverability_test.go

```diff
@@ -153,6 +160,7 @@ func enableMCSServiceDiscovery(_ resource.Context, cfg *istio.Config) {
 values:
   pilot:
     env:
+      PILOT_USE_ENDPOINT_SLICE: "true"
```

Comment:
Not sure if we should revert before merging or not? 

#34971 will make it so this runs in ES mode for k8s 1.21+ and EP mode for earlier.

Reply:
Let's revert in #34971. MCS relies on EndpointSlice anyway, so I think forcing this test to use EndpointSlice makes sense until your PR lands.